common/src/main/java/com/wynntils/models/character/CharacterModel.java

```diff
@@ -80,7 +84,7 @@ public final class CharacterModel extends Model {
     // This field is basically the slot id of the class,
     // meaning that if a class changes slots, the ID will not be persistent.
     // This was implemented the same way by legacy.
-    private int id;
+    private String id;
```

Comment:
Ideally, I'd like to see us correlate the value from the soul point tooltip with the full uuid from the API. But I guess it is "room for future improvements".

Reply:
I am not fully sure if that is needed though.

Reply:
Neither am I, but I'm fairly certain it's still a good idea. :-D

My point being, I can't figure out good reasons right now for having an UUID instead of this truncated string, but if you were to ask me to design this, if I did not know I was given just a partial UUID from the soul points, I would have chosen UUID as the representation.

And one reason to have the UUID is to match it with the data provided by Wynntils API. And if we have to match it with the UUID anyway, then we might just as well store it as UUID fully.